src/codewhisperer/commands/onInlineAcceptance.ts

```diff
@@ -82,10 +82,7 @@ export async function onInlineAcceptance(
          * Mitigation to right context handling mainly for auto closing bracket use case
          */
         try {
-            // Do not handle extra bracket if there is a right context merge
-            if (acceptanceEntry.recommendation === session.recommendations[acceptanceEntry.acceptIndex].content) {
```

Comment:
// Do not handle extra bracket if and only if there is a right context merge.

DO handle extra bracket if 
1. There is no right context merge, accepted recommendation is the same from service response.
2. There is no right context merge, accepted recommendation has some typeahead

Reply:
For 1, use condition `acceptanceEntry.recommendation === session.recommendations[acceptanceEntry.acceptIndex].content`

For 2, use session.recommendations[acceptanceEntry.acceptIndex].content.substring(0, typehead.length) === acceptanceEntry.recommendation`